File: reuse_kotlin/src/main/kotlin/Thrower.kt
/*
 * Classes reusability project.
 * Dac Linh Giao NGUYEN
 * License MIT
 *
 * Classes reusability in Kotlin.
 * Thrower.kt
 */
package com.kmmania

/**
 * Class representing a thrower, utilizing composition with the [Athlete] class.
 *
 * @property firstName The first name of the thrower.
 * @property lastName The last name of the thrower.
 * @property athlete An instance of the Athlete class representing the thrower's specialization.
 * @constructor Creates a Thrower with the specified first name, last name, and athlete instance.
 * @param firstName The first name of the thrower.
 * @param lastName The last name of the thrower.
 * @param athlete An instance of the Athlete class representing the thrower's specialization.
 * @author Dac Linh Giao NGUYEN
 * @since 0.1
 * @see Athlete
 */
class Thrower(
    val firstName: String,
    val lastName: String,
    private val athlete: Athlete
) {
    /**
     * Displays the information of the thrower including their name and speciality.
     */
    fun display() {
        println("My name is $firstName $lastName, ${athlete.mySpeciality()}")
    }
}

File: reuse_js/src/thrower.js
/**
 * Class representing a thrower, utilizing composition with the Athlete class.
 *
 * @author Dac Linh Giao NGUYEN
 * @version 0.1
 * @since 0.1
 */
class Thrower {
    /**
     * Create a thrower athlete.
     *
     * @param {string} firstName - The first name of the thrower.
     * @param {string} lastName - The last name of the thrower.
     * @param {Athlete} athlete - An instance of the Athlete class representing the thrower's specialization.
     */
    constructor(
        firstName,
        lastName,
        athlete
    ) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.athlete = athlete;
    }

    /**
     * Display thrower's information including name and speciality.
     */
    display() {
        console.log(`My name is ${this.firstName} ${this.lastName}. ${this.athlete.mySpeciality()}`);
    }
}

module.exports.Thrower = Thrower;